pkg/storage/bloom/v1/block.go

```diff
@@ -104,24 +103,24 @@ func (b *Block) Schema() (Schema, error) {
 }
 
 type BlockQuerier struct {
-	series *LazySeriesIter
+	*LazySeriesIter
```

Comment:
It now fulfills the `Iterator[Series]` of the `LazySeriesIter` rather than an iterator over `SeriesWithBloom` (v1 variant). Blooms (which can be in multiple pages) are now loaded independently and changing the signatures of this struct indicate that. This is done in two ways:
* Manually via `<BlockQuerier>.blooms.LoadOffset`. This give more manual control and is used by the `FusedQuerier` directly as well as by the new `offsetsIter`.
* By using the `<BlockQuerier>.Iter()` which returns a new impl of `Iterator[*SeriesWithBlooms]`. This is mostly convenience for testing/etc and allows iterating across a whole block, yielding `*SeriesWithBlooms`, but isn't used inside the `FusedQuerier` since we don't want to load every series on a query, only the requested ones.